backend/de.metas.postfinance/src/main/java/de/metas/postfinance/document/results/GetResultsFromPostFinanceService.java

```diff
@@ -0,0 +1,185 @@
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.postfinance.document.results;
+
+import com.google.common.collect.ImmutableList;
+import de.metas.attachments.AttachmentEntryCreateRequest;
+import de.metas.attachments.AttachmentEntryService;
+import de.metas.attachments.AttachmentTags;
+import de.metas.document.archive.DocOutboundLogId;
+import de.metas.document.archive.api.IDocOutboundDAO;
+import de.metas.document.archive.model.I_C_Doc_Outbound_Log;
+import de.metas.document.archive.model.X_C_Doc_Outbound_Log;
+import de.metas.organization.OrgId;
+import de.metas.postfinance.B2BServiceWrapper;
+import de.metas.postfinance.docoutboundlog.PostFinanceLog;
+import de.metas.postfinance.docoutboundlog.PostFinanceLogCreateRequest;
+import de.metas.postfinance.docoutboundlog.PostFinanceLogRepository;
+import de.metas.postfinance.document.export.PostFinanceExportException;
+import de.metas.postfinance.jaxb.DownloadFile;
+import de.metas.postfinance.processprotocol.BillType;
+import de.metas.postfinance.processprotocol.Envelope;
+import de.metas.util.Services;
+import de.metas.util.StringUtils;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.util.lang.impl.TableRecordReference;
+import org.springframework.stereotype.Service;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+import java.io.ByteArrayInputStream;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class GetResultsFromPostFinanceService
+{
+	@NonNull private final B2BServiceWrapper b2BServiceWrapper;
+	@NonNull private final PostFinanceLogRepository postFinanceLogRepository;
+	@NonNull private final AttachmentEntryService attachmentEntryService;
+
+	private final IDocOutboundDAO docOutboundDAO = Services.get(IDocOutboundDAO.class);
+
+	public void handleResultsFromPostFinance(@NonNull final OrgId orgId, @NonNull final TableRecordReference adPInstanceReference)
+	{
+		final List<DownloadFile> processProtocol = b2BServiceWrapper.getProcessProtocol(orgId);
+		processProtocol.forEach(file -> {
+
+			final AttachmentEntryCreateRequest attachmentEntryCreateRequest = createAttachmentRequest(file);
+			attachmentEntryService.createNewAttachment(adPInstanceReference, attachmentEntryCreateRequest);
+
+			try
+			{
+				JAXBContext jc = JAXBContext.newInstance(Envelope.class);
+				final Unmarshaller unmarshaller = jc.createUnmarshaller();
+				final ByteArrayInputStream inputStream = new ByteArrayInputStream(file.getData().getValue());
+
+				final Envelope envelope = (Envelope)unmarshaller.unmarshal(inputStream);
+
+				envelope.getBody().getDeliveryDate().forEach(deliveryDate -> handleDeliveryDate(deliveryDate, attachmentEntryCreateRequest, orgId));
```

Comment:
It may be better to only attach billType relevant to C_Doc_outboundlog instead of the complete file in  attachmentEntryCreateRequest

Reply:
As talked, it's ok to keep it like this